app/server/lib/AuditLogger.ts

```diff
@@ -105,6 +111,22 @@ export class AuditLogger implements IAuditLogger {
     this._subscribeToStreamingDestinations();
   }
 
+  public async close() {
+    this._installStreamingDestinations.clear();
+    this._orgStreamingDestinations.clear();
+    const listToClear = this._logsInProgress.slice();
+    this._logsInProgress = [];
+    for (const logCall of listToClear) {
+      // All log calls should be awaited already and errors should be logged.
+      await logCall.catch(() => {});
+    }
+    if (this._logsInProgress.length > 0) {
```

Comment:
Right, I understand why you do that: if the `logEventOrThrow` method is called while [lines 119 to 122 ](https://github.com/gristlabs/grist-core/pull/1292/files#diff-cba20aa17df9d34fa6e6df2671fd3514df5f3da8d5971aab72f805271332192cR119-R122)are executed, this should log an error (just like the message suggests).

But the tests might hang if such a case occur, don't they?

I have an alternative in mind, would it make sense?
 - we introduce a boolean property (`_isClosed` for example) which would indicate if the `close` method has been called;
 - in `logEventOrThrow()`, if the AuditLogger has been closed, we don't even call the internal method `_logEventOrThrow` and rather log the message below. This way, nothing is done and we don't have to worry about  what happens in this case anyway.